frontend/src/views/ViewSecretPublicPage/components/PasswordContainer.tsx

```diff
@@ -0,0 +1,85 @@
+import { useState, ChangeEvent } from "react";
+
+import { faArrowRight, faSpinner } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import { Button, IconButton, Input } from "@app/components/v2";
+import { fetchIsSecretPasswordValid } from "@app/hooks/api/secretSharing";
+import { createNotification } from "@app/components/notifications";
+
+type Props = {
+  secretId: string;
+  hashedHex: string;
+  handlePassMatch: (val: boolean) => void;
+};
+
+export const PasswordContainer = ({ secretId, hashedHex, handlePassMatch }: Props) => {
+  const [password, setPassword] = useState<string>('')

```

Comment:
Use react hook form for all form management. You can avoid keeping isLoading and everything

Reply:
Refactored to that. I had avoided it as state hooks require less code (in this case) as it's just one input.